WdtBase.cpp

```diff
@@ -163,6 +171,24 @@ WdtTransferRequest::WdtTransferRequest(const string& uriString) {
       }
     }
   } while (!portsList.empty());
+  string startPortStr = wdtUri.getQueryParam(START_PORT_PARAM);
+  string numPortsStr = wdtUri.getQueryParam(NUM_PORTS_PARAM);
+  if (!startPortStr.empty() && !numPortsStr.empty()) {
+    int32_t startPort = folly::to<int32_t>(startPortStr);
+    int numPorts = folly::to<int>(numPortsStr);
+    for (int i = 0; i < numPorts; i++) {
```

Comment:
loop as a service ? though semi seriously that code is already there somewhere (the wdtCmdLine uses it)